pkg/apis/openebs.io/upgrade/v1alpha1/upgradetask.go

```diff
@@ -0,0 +1,195 @@
+	// ImagePrefix contains the url prefix of the image url. This field is
+	// optional. If not present upgrade takes the previously present ImagePrefix.
+	ImagePrefix string `json:"imagePrefix" protobuf:"bytes,5,name=imagePrefix"`
+	// ImageTag contains the customized tag for ToVersion if any. This field is
+	// optional. If not present upgrade takes the ToVersion as the ImageTag
+	ImageTag string `json:"imageTag" protobuf:"bytes,6,name=imageTag"`
+}
+
+// Flags provides additional optional arguments
+type Flags struct {
+	Timeout int `json:"timeout,omitempty" protobuf:"varint,1,opt,name=resourceType"`
+}
+
+// ResourceSpec is the type of resource which is to be upgraded.
+// Exactly one of its members must be set.
+type ResourceSpec struct {
+	// JivaVolume contains the details of the jiva volume to be upgraded
+	JivaVolume *JivaVolume `json:"jivaVolume,omitempty" protobuf:"bytes,1,opt,name=jivaVolume"`
+	// CStorVolume contains the details of the cstor volume to be upgraded
+	CStorVolume *CStorVolume `json:"cstorVolume,omitempty" protobuf:"bytes,2,opt,name=cstorVolume"`
+	// CStorPool contains the details of the cstor pool to be upgraded
+	CStorPool *CStorPool `json:"cstorPool,omitempty" protobuf:"bytes,3,opt,name=cstorPool"`
+	// StoragePoolClaim contains the details of the storage pool claim to be upgraded
+	StoragePoolClaim *StoragePoolClaim `json:"storagePoolClaim,omitempty" protobuf:"bytes,4,opt,name=storagePoolClaim"`
+}
+
+// JivaVolume is the ResourceType for jiva volume
+type JivaVolume struct {
+	// PVName contains the name of the pv associated with the jiva volume
+	PVName string         `json:"pvName,omitempty" protobuf:"bytes,1,name=pvName"`
+	Flags  *ResourceFlags `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// CStorVolume is the ResourceType for cstor volume
+type CStorVolume struct {
+	// PVName contains the name of the pv associated with the cstor volume
+	PVName string         `json:"pvName,omitempty" protobuf:"bytes,1,name=pvName"`
+	Flags  *ResourceFlags `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// CStorPool is the ResourceType for cstor pool
+type CStorPool struct {
+	// PoolName contains the name of the cstor pool to be upgraded
+	PoolName string         `json:"poolName,omitempty" protobuf:"bytes,1,name=poolName"`
+	Flags    *ResourceFlags `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// StoragePoolClaim is the ResourceType for storage pool claim
+type StoragePoolClaim struct {
+	// SPCName contains the name of the storage pool claim to be upgraded
+	SPCName string         `json:"spcName,omitempty" protobuf:"bytes,1,name=spcName"`
+	Flags   *ResourceFlags `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// ResourceFlags provides additional options for a particular resource
+type ResourceFlags struct {
+	IgnoreStepsOnError []string `json:"ignoreStepsOnError,omitempty" protobuf:"bytes,1,opt,name=ignoreStepsOnError"`
```

Comment:
worth to add a comment on this

Reply:
comments please.
If step names will be pushed here. Then steps should be part of spec as well

Reply:
Added comments